util/cache/redis_hook.go

```diff
@@ -2,13 +2,14 @@ package cache
 
 import (
 	"context"
-	"strings"
+	"errors"
+	"net"
 
 	"github.com/redis/go-redis/v9"
 	log "github.com/sirupsen/logrus"
 )
 
-const NoSuchHostErr = "no such host"
+var dnsError *net.DNSError
```

Comment:
Is there any good reason to have this declared at the package global? If not, can you move this variable inside the function that depends on it and avoid the global state?